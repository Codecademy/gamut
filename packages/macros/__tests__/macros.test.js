'use strict';

const babel = require('@babel/core');
const path = require('path');

const options = {
  plugins: [require.resolve('babel-plugin-macros')],
};

const resolvePath = (relative) => path.resolve(__dirname, relative);

describe('title.macro', () => {
  test('should transform a js file', () => {
    const inputFile = resolvePath('../example/stories/World/Hello.js');

    const result = babel.transformFileSync(inputFile, options);

    expect(result.code).toMatchSnapshot();
  });

  test('should transform a story file', () => {
    const inputFile = resolvePath(
      '../example/stories/World/Example.stories.js'
    );

    const result = babel.transformFileSync(inputFile, options);

    expect(result.code).toMatchSnapshot();
  });
});
