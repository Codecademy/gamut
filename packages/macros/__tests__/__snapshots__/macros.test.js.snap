// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title.macro should transform a js file 1`] = `
"\\"use strict\\";

const meta = {
  default: \\"World/Hello\\",
  title: \\"World/Hello\\"
};"
`;

exports[`title.macro should transform a story file 1`] = `
"\\"use strict\\";

const meta = {
  default: \\"World/Example\\",
  title: \\"World/Example\\"
};"
`;
